src/state_machine/coordinator/fire.rs

```diff
@@ -943,91 +943,101 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
-                warn!(signer_id = %sig_share_response.signer_id, "No keys IDs configured");
-                return Ok(());
-            };
+        debug!(
+            "Sign round {} SignatureShareResponse for round {} from signer {} ({}/{} key_ids). Waiting on {:?}",
+            self.current_sign_id,
+            sig_share_response.sign_id,
+            sig_share_response.signer_id,
+            response_info.sign_recv_key_ids.len(),
+            response_info.nonce_recv_key_ids.len(),
+            response_info.sign_wait_signer_ids
+        );
+
+        if sig_share_response.dkg_id != self.current_dkg_id {
+            return Err(Error::BadDkgId(
+                sig_share_response.dkg_id,
+                self.current_dkg_id,
+            ));
+        }
+        if sig_share_response.sign_id != self.current_sign_id {
+            return Err(Error::BadSignId(
+                sig_share_response.sign_id,
+                self.current_sign_id,
+            ));
+        }
 
-            let mut sig_share_response_key_ids = HashSet::new();
-            for sig_share in &sig_share_response.signature_shares {
-                for key_id in &sig_share.key_ids {
-                    sig_share_response_key_ids.insert(*key_id);
-                }
-            }
+        // check that the signer_id exists in the config
+        let signer_public_keys = &self.config.signer_public_keys;
+        if !signer_public_keys.contains_key(&sig_share_response.signer_id) {
+            warn!(signer_id = %sig_share_response.signer_id, "No public key in config");
+            return Err(Error::MissingPublicKeyForSigner(
+                sig_share_response.signer_id,
+            ));
+        };
 
-            if *signer_key_ids != sig_share_response_key_ids {
-                warn!(signer_id = %sig_share_response.signer_id, "SignatureShareResponse key_ids didn't match config");
-                return Ok(());
+        // check that the key_ids match the config
+        let Some(signer_key_ids) = self
+            .config
+            .signer_key_ids
+            .get(&sig_share_response.signer_id)
+        else {
+            warn!(signer_id = %sig_share_response.signer_id, "No keys IDs configured");
+            return Err(Error::MissingKeyIDsForSigner(sig_share_response.signer_id));
+        };
+
+        let mut sig_share_response_key_ids = HashSet::new();
+        for sig_share in &sig_share_response.signature_shares {
+            for key_id in &sig_share.key_ids {
+                sig_share_response_key_ids.insert(*key_id);
             }
+        }
 
-            let have_shares = self
-                .signature_shares
-                .contains_key(&sig_share_response.signer_id);
+        if *signer_key_ids != sig_share_response_key_ids {
+            warn!(signer_id = %sig_share_response.signer_id, "SignatureShareResponse key_ids didn't match config");
+            return Err(Error::BadKeyIDsForSigner(sig_share_response.signer_id));
+        }
```

Comment:
qq: are we expecting `sig_share_response.signature_shares` to be a partition over the key ids controlled by the signer, or for each share to have all the key ids, or we don't care as long as it's a cover of the signer keys? This check enforces the last condition, but I wanted to double check this is what we wanted

Reply:
Oh, I see what you're saying.  This is a `v1` vs `v2` issue: in `v1` each SignatureShare can only be for one key.  So the only thing we can reasonably do here is make sure that a single SignatureShareResponse covers all keys for that signer.